pruning/architecture/utility/training.py

```diff
@@ -115,6 +116,55 @@ def validate_epoch(
     return metrics
 
 
+def setup_ddp(rank: int, world_size: int, init_method: str, seed: int = None) -> None:
+    """Setup the distributed training environment.
+
+    Args:
+        rank (int): The rank of the process.
+        world_size (int): The number of processes.
+        init_method (str): The initialization method.
+        seed (int, optional): The seed for reproducibility.
+    """
+
+    if seed is not None:
+        set_reproducibility(seed)
+
+    torch.cuda.set_device(rank)
+    dist.init_process_group(
```

Comment:
do we want to have hardcoded backend?

Reply:
Only `nccl` supports our code, there is also `gloo` but only for CPU and I don't think it would be useful.